main.tf

```diff
@@ -5,6 +5,7 @@ locals {
   ecr_repo_name    = local.app_name_and_env
   is_multiregion   = var.aws_region_secondary != ""
   is_primary       = local.is_multiregion && var.aws_region != var.aws_region_secondary
+  create_cd_user   = !local.is_multiregion || local.is_primary
```

Comment:
You don't need to change this, and maybe I got it wrong.  I just feeling like "sharing" this.

I made a chart of the different possible scenarios with these inter-dependent variables.
Assuming, `var.aws_region` will never be blank, it looks to me like you can just say `create_cd_user = var.aws_region != var.aws_region_secondary`

Reply:
I guess this brings up a question in my mind in relation to the variable definition below.

Do we really want to make the code assume there will always be a secondary region? Wouldn't it be better to make it optional and also explicit (e.g. by making `is_multiregion` and `is_primary` vars rather than locals)? 

```
variable "aws_region_secondary" {
  description = "AWS region in which to create replica resources. If omitted, no replicas are created and this hub is configured to be the secondary."
  type        = string
  default     = ""
}
```

Is this saying that if you want a single region setup, then you have to make `aws_region_secondary` equal to `aws_region`?

Reply:
You're right. `var.aws_region != var.aws_region_secondary` is equivalent to `!local.is_multiregion || local.is_primary` but my desire was to use the locals to more clearly describe what's happening. Did you go create your logic diagram with the goal of reviewing the code or because it didn't make sense?

> Is this saying that if you want a single region setup, then you have to make aws_region_secondary equal to aws_region?
That's not what I intended. In a single-region setup, you can omit `aws_region_secondary`. I'll add that detail to the description.

Reply:
I think the line on its own makes sense.
It's just that the more I looked at the whole set of related locals, the more I thought I should check it out for PR review purposes.